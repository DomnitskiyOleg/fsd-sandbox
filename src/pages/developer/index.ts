export { DeveloperPage } from './ui/Page'
