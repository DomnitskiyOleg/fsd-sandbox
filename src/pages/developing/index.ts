export { DevelopingPage } from './ui/Page'
