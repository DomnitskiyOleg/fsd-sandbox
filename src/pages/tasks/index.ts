export { TasksPage } from './ui/Page'
