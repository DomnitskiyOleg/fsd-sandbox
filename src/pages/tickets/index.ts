export { TicketsPage } from './ui/Page'
