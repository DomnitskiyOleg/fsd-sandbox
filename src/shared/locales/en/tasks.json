{
  "title": "Tasks",
  "AddTaskForm": {
    "nameLabel": "Task name",
    "descriptionLabel": "Description",
    "dateLabel": "Complete by",
    "addButton": "Add",
    "resetButton": "Reset"
  },
  "TaskCard": {
    "noDescription": "No description",
    "completeButton": "Complete",
    "startButton": "Start",
    "deleteButton": "Delete"
  }
}
