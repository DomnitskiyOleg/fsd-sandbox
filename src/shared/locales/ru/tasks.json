{
  "title": "Tasks",
  "AddTaskForm": {
    "nameLabel": "Название задачи",
    "descriptionLabel": "Описание",
    "dateLabel": "Выполнить до",
    "addButton": "Добавить",
    "resetButton": "Сбросить"
  },
  "TaskCard": {
    "noDescription": "Нет описания"
  }
}
