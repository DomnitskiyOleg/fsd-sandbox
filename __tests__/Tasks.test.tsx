import userEvent from '@testing-library/user-event'
import { screen, within } from '@testing-library/dom'
import { initI18n } from './mocks'
import { render } from './mocks'
import { TasksPage } from '@/pages/tasks'

beforeAll(async () => {
    await initI18n()
})

describe('Tasks page testing', () => {
    test('renders AddForm', () => {
        render(<TasksPage />)

        const addButton = screen.getByRole('button', { name: /add/i })
        expect(addButton).toBeInTheDocument()
    })

    test('adds task with title', async () => {
        render(<TasksPage />)

        const titleInput = screen.getByRole('textbox', { name: /name/i })
        const addButton = screen.getByRole('button', { name: /add/i })

        await userEvent.type(titleInput, 'test task1')
        await userEvent.click(addButton)

        const task = screen.getByText(/test task1/i)

        expect(task).toBeInTheDocument()
    })

    test('full form filling', async () => {
        const { asFragment } = render(<TasksPage />)

        const addButton = screen.getByRole('button', { name: /add/i })
        const titleInput = screen.getByRole('textbox', { name: /name/i })
        const desInput = screen.getByRole('textbox', { name: /description/i })
        const dateInput = screen.getByRole('textbox', { name: /complete by/i })

        await userEvent.type(titleInput, 'test task2')
        await userEvent.type(desInput, 'example description')
        await userEvent.type(dateInput, '12/17/2024')
        await userEvent.click(addButton)

        const task = screen.getByText(/test task2/i)
        const taskCard = task.closest('.task-card')

        expect(task).toBeInTheDocument()
        expect(taskCard).toBeInTheDocument()
        expect(taskCard).toContainElement(
            screen.getByText(/example description/i),
        )
        expect(taskCard).toContainElement(screen.getByText('17/12/2024'))
        expect(asFragment()).toMatchSnapshot()
    })

    test('filters tasks', async () => {
        render(<TasksPage />)

        const task1 = screen.getByText(/test task1/i)
        const task2 = screen.getByText(/test task2/i)

        const doneFilterButton = screen.getByRole('button', { name: /done/i })
        await userEvent.click(doneFilterButton)

        expect(task1).not.toBeInTheDocument()
        expect(task2).not.toBeInTheDocument()
    })

    test('starts tasks', async () => {
        render(<TasksPage />)

        const allFilterButton = screen.getByRole('button', { name: /all/i })
        const inProgressFilterButton = screen.getByRole('button', {
            name: /in progress/i,
        })

        await userEvent.click(allFilterButton)

        const task1 = screen.getByText(/test task1/i)
        const task1Card = task1.closest('.task-card') as HTMLElement

        const startButton = within(task1Card).getByRole('button', {
            name: /start/i,
        })

        await userEvent.click(startButton)
        await userEvent.click(inProgressFilterButton)

        expect(task1).toBeInTheDocument()
    })

    test('deletes tasks', async () => {
        render(<TasksPage />)

        const allFilterButton = screen.getByRole('button', { name: /all/i })
        await userEvent.click(allFilterButton)

        const task1 = screen.getByText(/test task1/i)
        const task2 = screen.getByText(/test task2/i)

        const task1Card = task1.closest('.task-card') as HTMLElement
        const task2Card = task2.closest('.task-card') as HTMLElement

        const deleteButton1 = within(task1Card).getByRole('button', {
            name: /delete/i,
        })
        const deleteButton2 = within(task2Card).getByRole('button', {
            name: /delete/i,
        })

        await userEvent.click(deleteButton1)
        await userEvent.click(deleteButton2)

        expect(task1).not.toBeInTheDocument()
        expect(task2).not.toBeInTheDocument()
    })
})
