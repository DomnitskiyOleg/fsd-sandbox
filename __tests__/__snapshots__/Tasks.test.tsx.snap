// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks page testing renders the heading Vite + React 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-1gh65au-MuiContainer-root"
  >
    <form>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 css-1fdopfm-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-ubb9a5-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-12 MuiGrid2-grid-lg-3 css-15qsftl-MuiGrid2-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1kfabtt-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Task name
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="name"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-81qg8w"
                  >
                    <span>
                      Task name *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-8 MuiGrid2-grid-lg-6 css-1x9fed0-MuiGrid2-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1kfabtt-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r1:"
                id=":r1:-label"
              >
                Description
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  name="description"
                  rows="2"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-81qg8w"
                  >
                    <span>
                      Description
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-4 MuiGrid2-grid-lg-3 css-1wrx4wi-MuiGrid2-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-xwn1oi-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1kfabtt-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                Complete by
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-1n04w30-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-13meb6w-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  inputmode="text"
                  placeholder="MM/DD/YYYY"
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-elo8k2-MuiInputAdornment-root"
                >
                  <button
                    aria-label="Choose date"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-oeq6yz-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CalendarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                    </svg>
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-81qg8w"
                  >
                    <span>
                      Complete by
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-1ndfs4l-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1btmbba-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Reset
            <span
              class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-pyt3b0-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="RestartAltOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 13c0-1.65.67-3.15 1.76-4.24L6.34 7.34C4.9 8.79 4 10.79 4 13c0 4.08 3.05 7.44 7 7.93v-2.02c-2.83-.48-5-2.94-5-5.91m14 0c0-4.42-3.58-8-8-8-.06 0-.12.01-.18.01l1.09-1.09L11.5 2.5 8 6l3.5 3.5 1.41-1.41-1.08-1.08c.06 0 .12-.01.17-.01 3.31 0 6 2.69 6 6 0 2.97-2.17 5.43-5 5.91v2.02c3.95-.49 7-3.85 7-7.93"
                />
              </svg>
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Add
            <span
              class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-pyt3b0-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="AddBoxOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 16H5V5h14zm-8-2h2v-4h4v-2h-4V7h-2v4H7v2h4z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-1gh65au-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal MuiToggleButtonGroup-fullWidth css-1hkie7k-MuiToggleButtonGroup-root"
        role="group"
      >
        <button
          aria-pressed="true"
          class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-firstButton css-1yjtblt-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="all"
        >
          All
        </button>
        <button
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-1yjtblt-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="todo"
        >
          To do
        </button>
        <button
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-middleButton css-1yjtblt-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="inProgress"
        >
          In progress
        </button>
        <button
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButtonGroup-lastButton css-1yjtblt-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="done"
        >
          Done
        </button>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-1 css-1bwoeeg-MuiGrid2-root"
    />
  </div>
</DocumentFragment>
`;
